main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java

```diff
@@ -85,23 +86,175 @@ private <E extends GtfsEntity> int loadUniqueCount(
   }
 
   private void loadSpecFeatures(GtfsFeedContainer feedContainer) {
+    loadSpecFeaturesBasedOnFilePresence(feedContainer);
+    loadSpecFeaturesBasedOnFieldPresence(feedContainer);
+  }
+
+  private void loadSpecFeaturesBasedOnFilePresence(GtfsFeedContainer feedContainer) {
+    List<Pair<String, String>> componentList =
+        List.of(
+            new Pair<>("Pathways", GtfsPathway.FILENAME),
+            new Pair<>("Transfers", GtfsTransfer.FILENAME),
+            new Pair<>("Fares V1", GtfsFareAttribute.FILENAME),
+            new Pair<>("Fares V2", GtfsFareProduct.FILENAME),
+            new Pair<>("Shapes", GtfsShape.FILENAME),
+            new Pair<>("Frequency-Based Trip", GtfsFrequency.FILENAME),
+            new Pair<>("Feed Information", GtfsFeedInfo.FILENAME),
+            new Pair<>("Attributions", GtfsAttribution.FILENAME),
+            new Pair<>("Translations", GtfsTranslation.FILENAME),
+            new Pair<>("Fare Media", GtfsFareMedia.FILENAME),
+            new Pair<>("Zone-Based Fares", GtfsStopArea.FILENAME),
+            new Pair<>("Transfers", GtfsTransfer.FILENAME));
+
+    for (Pair<String, String> entry : componentList) {
+      specFeatures.put(
+          entry.getKey(),
+          hasAtLeastOneRecordInFile(feedContainer, entry.getValue()) ? "Yes" : "No");
+    }
+  }
+
+  private void loadSpecFeaturesBasedOnFieldPresence(GtfsFeedContainer feedContainer) {
```

Comment:
I think there is an opportunity here to reduce the number of lines for the method following the same pattern as before having a single method per component. Having shorter methods improve the readability and maintainability of the code.

Reply:
just to make sure i understand properly, you suggest we separate this method in multiple small methods (one per component) for improved readability?

Reply:
Yes, that's the suggestion.